[TOPIC-BLE-LLCP] EDTT integration

thoh-ot opened the pull request:
This PR adds:

- building both new and legacy variant of hci_test_app
- fine grained verbosity
- rr debug support

thoh-ot committed:
tests: Bluetooth: edtt: Add per program verbosity

Add more fine grained control of the verbosity with new environment
variables:

VERBOSITY_LEVEL: Global verbosity [2]
VERBOSITY_LEVEL_EDTT: EDTT verbosity [VERBOSITY_LEVEL]
VERBOSITY_LEVEL_BRIDGE: EDTT bridge verbosity [VERBOSITY_LEVEL]
VERBOSITY_LEVEL_PHY: bsim phy verbosity [VERBOSITY_LEVEL]
VERBOSITY_LEVEL_DEVS: Global bsim device verbosity [VERBOSITY_LEVEL]
VERBOSITY_LEVEL_DEV1: bsim device 1 verbosity [VERBOSITY_LEVEL_DEVS]
VERBOSITY_LEVEL_DEV1: bsim device 2 verbosity [VERBOSITY_LEVEL_DEVS]

Value in [] is the default verbosity unless overridden.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
tests: Bluetooth: edtt: Add per device project configuration

PRJ_CONF: Default bsim device configuration [prj_conf]
PRJ_CONF_1: bsim device 1 configuration [PRJ_CONF]
PRJ_CONF_2: bsim device 2 configuration [PRJ_CONF]

Value in [] is the default unless overridden.

Configuration names should match those used in compile.sh.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
tests: Bluetooth: edtt: Add rr debug support

RR: Default run bsim device under rr [0]
RR_1: Run bsim device 1 under rr [RR]
RR_2: Run bsim device 2 under rr [RR]

Value in [] is the default unless overridden.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
tests: Bluetooth: bsim: Add better debug support to compile

Enable debug optimization and cmake compile commands generation.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

aescolar reviewed (commented):
This seems overall ok. Most of these commits could target master I guess?

thoh-ot commented:
> This seems overall ok. Most of these commits could target master I guess?

Yes. Only two commits are specific to the LLCP work.

thoh-ot committed:
tests: Bluetooth: edtt: Fix compilation error

Refactored LLCP controller does not support Advanced Scheduling yet.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

erbr-ot reviewed (approved):
Looks sweet - nice!

kruithofa commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/prj_legacy.conf:
```diff
@@ -0,0 +1,27 @@
+
+CONFIG_SYS_POWER_MANAGEMENT=y
+CONFIG_TICKLESS_IDLE=y
+CONFIG_TICKLESS_KERNEL=y
```
At some point in the future the default LL will switch to the refactored LLCP. Wouldn't it be better to explicitly select the legacy LLCP here:

`# Disabling BT_LL_SW_SPLIT_LLCP_LEGACY requires BT_CTLR_ADVANCED_FEATURES`
`CONFIG_BT_CTLR_ADVANCED_FEATURES=y`
`CONFIG_BT_LL_SW_SPLIT_LLCP_LEGACY=y`

Even though legacy is default now, when changing the default it is easy to overlook that this file needs to be changed at that time.

thoh-ot replied:
ACK

kruithofa commented:
It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?

kruithofa commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/_controller_tests_inner.sh:
```diff
@@ -4,16 +4,32 @@
+    local exe=$(basename $1)
+    local out=/tmp/rr/$$-${SIMULATION_ID}-${exe}-d_${devno}
+    rm -rf ${out}
+    rr="rr record -o ${out}"
```
if we are paranoid we could add a check to see if rr is installed

thoh-ot replied:
ACK

thoh-ot committed:
tests: Bluetooth: bsim: Build multiple variants of hci_test_app

Build both the new and legacy variant of the hci_test_app.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot commented:
> It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?

In the PR description or somewhere else?

kruithofa reviewed (approved):
LGTM

kruithofa commented:
> > It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?
> 
> In the PR description or somewhere else?

I think in the script itself would be best

thoh-ot committed:
tests: Bluetooth: edtt: Add config documentation

Document the different environment variables that can alter the
behaviour of the script.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot commented:
> > > It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?
> > 
> > 
> > In the PR description or somewhere else?
> 
> I think in the script itself would be best

I have added some top level documentation of the environment variables in the script.
